dps-common/caasauth/caasauth.go

```diff
@@ -0,0 +1,254 @@
+	req, err := http.NewRequest("POST", identityUrl, bytes.NewReader(reqBody))
+	if err != nil {
+		return nil, err
+	}
+	//req.Header.Add("Accept", "application/json")
+	req.Header.Add("Content-Type", "application/json")
+
+	c := &http.Client{
+		Transport: LoggedTransportConfig,
+	}
+
+	resp, err := c.Do(req)
+
+	if resp == nil && err == nil {
+		return nil, errors.New(fmt.Sprintf("Unknown error while connnecting to %s", identityUrl))
+	}
+
+	if resp == nil || resp.StatusCode >= 400 {
+		if err != nil {
+			return nil, err
+		}
+		return nil, errors.New(fmt.Sprintf("Error while connnecting to %s, please check your username and password or use flags for each command. (%s)", identityUrl, resp.Status))
+	}
+
+	var tokenResp TokenResponse
+	body, _ := ioutil.ReadAll(resp.Body)
+	err = json.Unmarshal(body, &tokenResp)
+
+	return &tokenResp, nil
+}
+
+func cutAndTrimAddress(address string) (string, string) {
+	for _, v := range PREFIX_TRIM {
+		address = strings.TrimPrefix(address, v)
+	}
+	address = strings.TrimRight(address, "/ ")
+	basePath := ""
+	slashIndex := strings.Index(address, "/")
+	if slashIndex != -1 {
+		basePath = address[slashIndex:]
+		address = address[0:slashIndex]
+	}
+	return address, basePath
+}
+
+type Transport struct {
+	Runtime *client.Runtime
+}
+
+func (t *Transport) Submit(operation *runtime.ClientOperation) (interface{}, error) {
+	operation.Reader = &noContentSafeResponseReader{OriginalReader: operation.Reader}
+	response, err := t.Runtime.Submit(operation)
+	return response, err
+}
+
+type noContentSafeResponseReader struct {
+	OriginalReader runtime.ClientResponseReader
+}
+
+type ErrorMessage struct {
+	Message         string            `json:"message"`
+	ValidationError map[string]string `json:"validationErrors"`
+}
+
+func (e *ErrorMessage) String() string {
+	var result string = ""
+	if len(e.Message) > 0 {
+		result = e.Message
+	} else if len(e.ValidationError) > 0 {
+		var validationErrors []string
+		for k, v := range e.ValidationError {
+			message := fmt.Sprintf("'%v' - %v", strings.TrimPrefix(k, "postPrivate.arg0."), v)
+			validationErrors = append(validationErrors, message)
+		}
+		result = strings.Join(validationErrors, ",")
+	}
+	return result
+}
+
+func (r *noContentSafeResponseReader) ReadResponse(response runtime.ClientResponse, consumer runtime.Consumer) (interface{}, error) {
+	resp, err := r.OriginalReader.ReadResponse(response, consumer)
+	if err != nil {
+		switch response.Code() {
+		case 200:
+		case 202:
+		case 204:
+			return nil, nil
+		default:
+			body, _ := ioutil.ReadAll(response.Body())
+			var errorMessage ErrorMessage
+			err := json.Unmarshal(body, &errorMessage)
+			if err != nil || (len(errorMessage.Message) == 0 && len(errorMessage.ValidationError) == 0) {
+				return nil, &utils.RESTError{Response: string(body), Code: response.Code()}
+			}
+			return nil, &utils.RESTError{Response: errorMessage.String(), Code: response.Code()}
+		}
+	}
+	return resp, nil
```

Comment:
Missing error handling or error out sourcing. I suggest `return resp, err` otherwise the error is simply disappear.

Reply:
In this case, it won't make any difference

Reply:
but bad practice to ignore any kind of error